go-mode.el

```diff
@@ -1503,7 +1506,9 @@ description at POINT."
         (if (not (godef--successful-p file))
             (message "%s" (godef--error file))
           (push-mark)
-          (ring-insert find-tag-marker-ring (point-marker))
+          (if (eval-when-compile (fboundp 'xref-push-marker-stack))
+              (xref-push-marker-stack)
```

Comment:
Not in the scope of this PR, but we should probably hook into xref properly, so that `xref-find-definitions` works as expected.